client/app/components/form/widgets/FileWidget.tsx

```diff
@@ -109,11 +163,20 @@ const FileWidget = ({
   const handleChange = useCallback(
     (event: ChangeEvent<HTMLInputElement>) => {
       event.preventDefault();
-      if (!event.target.files) {
+      const maxSize = 1024 * 1024; // 1 MB in bytes
+      const files = event.target.files;
+      if (!files) {
         return;
       }
-      processFiles(event.target.files).then((filesInfoEvent) => {
-        const newValue = filesInfoEvent.map((fileInfo) => fileInfo.dataURL);
+
+      processFiles(files).then((filesInfoEvent) => {
+        const newValue = filesInfoEvent.map((fileInfo) => {
+          if (fileInfo.size > maxSize) {
+            alert("File size must be less than 1MB");
```

Comment:
Error handling wasn't part of this ticket and we don't have a design for it, so in the meantime I added this so we don't get unhandled errors

Reply:
Went over this with Nan, it's fine for now